src/analyses/wrapperFunctionAnalysis.ml

```diff
@@ -133,17 +133,20 @@ module MallocWrapper : MCPSpec = struct
   let query (ctx: (D.t, G.t, C.t, V.t) ctx) (type a) (q: a Q.t): a Q.result =
     let wrapper_node, counter = ctx.local in
     match q with
-    | Q.HeapVar ->
+    | Q.HeapVar {on_stack = on_stack} ->
       let node = match wrapper_node with
         | `Lifted wrapper_node -> wrapper_node
         | _ -> node_for_ctx ctx
       in
       let count = UniqueCallCounter.find (`Lifted node) counter in
       let var = NodeVarinfoMap.to_varinfo (ctx.ask Q.CurrentThreadId, node, count) in
       var.vdecl <- UpdateCil.getLoc node; (* TODO: does this do anything bad for incremental? *)
+      if on_stack then var.vattr <- addAttribute (Attr ("stack_alloca", [])) var.vattr; (* If the call was for stack allocation, add an attr to mark the heap var *)
       `Lifted var
     | Q.IsHeapVar v ->
       NodeVarinfoMap.mem_varinfo v
+    | Q.IsDynamicallyAlloced v ->
+      NodeVarinfoMap.mem_varinfo v && not @@ hasAttribute "stack_alloca" v.vattr
```

Comment:
In what sense are stack things allocated by `alloca` not dynamically allocated? I think the terminology is a bit confusing here.

Reply:
You're right, `alloca` is indeed dynamic allocation as well. I got `HeapVar` and `IsDynamicallyAlloced` the other way around.

Reply:
`IsHeapVar` and `IsDynamicallyAlloced` should be now implemented the other way around. All of their usages in Goblint should be also adapted to these changes now. Cf. https://github.com/goblint/analyzer/pull/1179/commits/d9df607e99ea78600b81f879955208ec00391bcb